### PR TITLE
ItemList: Clarify distinction between disabled and selected in sending signals

### LINKED ISSUE #74086: ItemList: Disabled items still trigger selection and activation signals on mouse
### Godot version

4.0.rc6

### System information

Windows 10

### Issue description

Disabled items of an `ItemList` still trigger `item_selected` and `item_activated` signals when using the mouse. Even though the documentation states that:
> Disabled items cannot be selected and do not trigger activation signals (when double-clicking or pressing Enter).

Visually, they are as expected (grayed out and can't grab focus).

-----
This seems to be a recurring problem as it has happened in Godot 2 (#5683) and Godot 3 (#37277) as well.
According to the comments in #37277, PR #37285 (#60123) is supposed to fix it for Godot 4. But it still seems to be an issue.

The PR suggests using `disabled` for rendering and `selectable` for actual selection. If that's the final verdict on the matter, then the documentation should be updated to reflect this.

### Steps to reproduce

1. Create a simple scene.
2. Add an `ItemList` and add several items to it via the Inspector. Disable one or more items.
3. Attach a script to the item list with the following code:
```gdscript
extends ItemList


func _ready() -> void:
	item_activated.connect(_on_item_activated)
	item_selected.connect(_on_item_selected)


func _on_item_activated(item_index: int) -> void:
	prints("Item Activated:", item_index)


func _on_item_selected(item_index: int) -> void:
	prints("Item Selected:", item_index)
```
4. Run the project and try selecting/activating a disabled item.
5. Check the Output console for results.

### Minimal reproduction project

[ItemList_DisabledItem_Bug.zip](https://github.com/godotengine/godot/files/10844570/ItemList_DisabledItem_Bug.zip)

### PR DESCRIPTION
**Pull request by daBlesr**
I have made sure there is a clear distinction in which situations signals are triggered:

when item is disabled: send no signal at all for this item (activated, selected, deselected, ...) 
when item is selectable: send signals (item_selected, multi_selected) when not disabled.

when is activated now triggered? When item is not disabled and double clicked (pressing enter is only possible when selectable 😄 )

I am in conflict with deselecting however, I am allowing deselection even when selectable is turned off to keep this behaviour as before.

Fixes: #74086

### CODE REVIEW

**Comment by daBlesr** [comment]
Hi @KoBeWi could I request a review from you, or @akien-mga could you add anyone as a reviewer whom you might think would be able to review this PR?

File: scene/gui/item_list.cpp [hunk 1]
```diff
@@ -661,18 +661,26 @@ void ItemList::gui_input(const Ref<InputEvent> &p_event) {
...
+				// Don't emit any signal or do any action with clicked item when disabled.
+				return;
+			}
+
 			if (select_mode == SELECT_MULTI && items[i].selected && mb->is_command_or_control_pressed()) {
 				deselect(i);
 				emit_signal(SNAME("multi_selected"), i, false);
 
 			} else if (select_mode == SELECT_MULTI && mb->is_shift_pressed() && current >= 0 && current < items.size() && current != i) {
+				// range selection
```
Comment (YuriSizov) [inline scene/gui/item_list.cpp:674]:
```suggestion
				// Range selection.
```

If you are adding code comments, please follow the proper style.

File: scene/gui/item_list.cpp [hunk 2]
```diff
@@ -661,18 +661,26 @@ void ItemList::gui_input(const Ref<InputEvent> &p_event) {
...
+
 				int from = current;
 				int to = i;
 				if (i < current) {
 					SWAP(from, to);
 				}
 				for (int j = from; j <= to; j++) {
 					if (!CAN_SELECT(j)) {
+						// item is not selectable during a range selection, so skip it.
```
Comment (YuriSizov) [inline scene/gui/item_list.cpp:683]:
```suggestion
						// Item is not selectable during a range selection, so skip it.
```

File: scene/gui/item_list.cpp [hunk 3]
```diff
@@ -708,7 +721,8 @@ void ItemList::gui_input(const Ref<InputEvent> &p_event) {
 
 			return;
 		} else if (closest != -1) {
-			emit_signal(SNAME("item_clicked"), closest, get_local_mouse_position(), mb->get_button_index());
+			if (!items[closest].disabled)
+				emit_signal(SNAME("item_clicked"), closest, get_local_mouse_position(), mb->get_button_index());
```
Comment (YuriSizov) [inline scene/gui/item_list.cpp:725]:
```suggestion
			if (!items[closest].disabled) {
				emit_signal(SNAME("item_clicked"), closest, get_local_mouse_position(), mb->get_button_index());
			}
```

You need to use full syntax for conditional statements.

**Review by KoBeWi — APPROVED** [review]
On technical side it looks good.

**Comment by akien-mga** [comment]
OP was no longer responsive, so I went ahead and fixed the style issues myself.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**